core/src/test/scala/unit/kafka/admin/AclCommandTest.scala

```diff
@@ -324,12 +348,18 @@ class AclCommandTest extends QuorumTestHarness with Logging {
   }
 
   private def withAuthorizer()(f: Authorizer => Unit): Unit = {
-    val kafkaConfig = KafkaConfig.fromProps(brokerProps, doLog = false)
-    val authZ = new AclAuthorizer
-    try {
-      authZ.configure(kafkaConfig.originals)
-      f(authZ)
-    } finally authZ.close()
+    if (isKRaftTest()) {
+      (servers.map(_.authorizer.get) ++ controllerServers.map(_.authorizer.get)).foreach { auth =>
```

Comment:
It seems to me those code works well with zk, so we can simplify `withAuthorizer`.
```scala
  private def withAuthorizer()(f: Authorizer => Unit): Unit = {
    (servers.map(_.authorizer.get) ++ controllerServers.map(_.authorizer.get)).foreach { auth =>
      f(auth)
    }
  }
```

Reply:
I've simplified the `withAuthorizer()` method with the implementation above. But there seems to be an issue with test methods like `testAclCliWithAuthorizer()`, `testProducerConsumerCliWithAuthorizer()`, `testAclsOnPrefixedResourcesWithAuthorizer()`, `testPatternTypesWithAuthorizer()`. In those methods, there are no running server or controller, so the above code in `withAuthorizer()` just does not execute anything at all, because the collection `(servers.map(_.authorizer.get) ++ controllerServers.map(_.authorizer.get))` is empty. 

Those tests pass, but `withAuthorizer()` doesn't check anything for them. So probably we are getting a bit less code coverege in this case.

Not sure if it's a big issue, as we are going to delete those test methods after full migration to KRaft anyway?

Reply:
> Those tests pass, but withAuthorizer() doesn't check anything for them. So probably we are getting a bit less code coverege in this case.

that is a good point. in zk mode we can modify the data through zk client, and hence it can work even though there is no broker. In order to test authorizer by interface (`KafkaBroker`), maybe we can create a broker for those tests after changing the data of zk. WDYT?

Reply:
Would it be a bit confusing for someone who looks at `testAclCliWithAuthorizer` and similar tests, if we start broker just for verifying ACLs in Zk?
For me, the original implementation of `withAuthorizer()` with extra check `if (isKRaftTest())` looks fine... at least it would be easy to remove the `else` branch, once we get rid of Zk - we'll just look for all places where `isKRaftTest()` is used in this test class

To optimise it a bit, maybe we can create and configure this `new AclAuthorizer` when test starts, e.g. in `setUp()` method. Something like this:

```
setUp() { ...
  aclAutorizerForZk = new AclAuthorizer
  aclAutorizerForZk.configure(..)
}

private def withAuthorizer()(f: Authorizer => Unit): Unit = {
    if (isKRaftTest()) {
      (servers.map(_.authorizer.get) ++ controllerServers.map(_.authorizer.get)).foreach { auth =>
        f(auth)
      }
    } else {
        f(aclAutorizerForZk)
    }
}
```

Reply:
My point was the test should honor `KafkaBroker` interface instead of creating/configuring the authorizer manually. However, your comment makes sense to me that zk is going to be dropped. My origin idea is over-engineering.

> To optimise it a bit, maybe we can create and configure this new AclAuthorizer when test starts, e.g. in setUp() method. Something like this:

That will create zk authorizer even though the tests don't have zk, right? 

> the original implementation of withAuthorizer() with extra check if (isKRaftTest()) looks fine... at least it would be easy to remove the else branch, once we get rid of Zk - we'll just look for all places where isKRaftTest() is used in this test class

let's keep original implementation :)

Reply:
> That will create zk authorizer even though the tests don't have zk, right?

Yeah, you are right...

Restored the original implementation :ok_hand: